### PR TITLE
Builds table: replace some project and jobset references in the Projects and Builds controllers

### PR DESCRIPTION
**Pull request by grahamc**
Extracted from #1093.

My methodology here is to:

1. Pick a commit from #1093
2. Check out the parent commit
3. Run the test suite and identify failing tests
4. Check out the picked commit and run those failing tests
5. Verify the picked commit fixed at least one test.

If it didn't, write a test and verify it is broken before / fixed after the picked commit.

The picked commit in this case was `Builds: LatestSucceeded*`.

### CODE REVIEW

File: t/Hydra/Controller/Job/builds.t [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+my $builds = $ctx->makeAndEvaluateJobset(
+    expression => "basic.nix",
+    build => 1
+);
+
+my $build = $builds->{"empty_dir"};
+my $jobset = $build->jobset;
+my $project = $build->project;
+
+subtest "/job/PROJECt/JOBSET/JOB/all" => sub {
```
Comment (cole-h) [inline t/Hydra/Controller/Job/builds.t:21]: Minor nit: any reason why the final `t` in `PROJECt` is lower-case?

File: t/Hydra/Controller/Job/builds.t [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    my $response = request(GET '/job/' . $project->name . '/' . $jobset->name . '/' . $build->job . '/all');
+    ok($response->is_success, "The page showing the job's builds returns 200.");
+};
+
+subtest "/job/PROJECt/JOBSET/JOB/channel/latest" => sub {
```
Comment (cole-h) [inline t/Hydra/Controller/Job/builds.t:26]: Here, too.

**Review by cole-h — COMMENTED** [review]
2 very minor nits, otherwise the diffs and tests all look sane to me.

File: t/Hydra/Controller/Job/builds.t (hunk 1 above)
Comment (grahamc) [reply t/Hydra/Controller/Job/builds.t:21]: Typo + copypasta I'm sure!

**Pull request merged**